File: Python/Easy/linked_list/remove_linked_list_element.py
"""

Given the head of a linked list and an integer val, 
remove all the nodes of the linked list that has Node.val == val, 
and return the new head.

"""

class ListNode:
    def __init__(self, val=0, next=None):
        self.val=val
        self.next=next

class Solution:
    def removeElements(self, head, val):
        while head and head.val==val:
            head=head.next #Checking edge case where first value == val
        a=head
        while a is not None:
            if a.next is not None and a.next.val==val:
                a.next=a.next.next
            else:
                a=a.next
        
        return head
    
    

File: JavaScript/Easy/linked_lists/remove_linked_list_element.js
/**
 * @param {ListNode} head
 * @return {head}
 */

var removeElements = function (head, val) {
    while (head && head.val==val){
        head=head.next
    }
    let a = head;
    while (a != null){
        if (a.next != null && a.next.val==val){
            a.next=a.next.next;
        } else {a=a.next}
    }
    return head
};

removeE => {
    return 100+200;
};

a => 100+200;